app/models/user.rb

```diff
@@ -75,6 +78,28 @@ def remove_devise_validations
     end
   end
 
+  def self.fetch_leader_board_stats(current_user)
```

Comment:
Just for posterity:

I think two faster queries with no temp space allocation here will be more efficient:

```ruby
top_ten = User.order('wallet_balance desc').limit(10)
current_rank = User.count('wallet_balance > ?', current_user.wallet_balance)

{top_ten: top_den, current_rank: current_rank}
```